consensus/leader_election/vrf/vrf.go

```diff
@@ -0,0 +1,98 @@
+package vrf
+
+// This file is a light wrapper around https://pkg.go.dev/github.com/ProtonMail/go-ecvrf.
+// It is needed to to achieve more semantic variable naming in the use of HotPocket.
+
+import (
+	"bytes"
+	"crypto/sha512"
+	"fmt"
+	"io"
+
+	"github.com/pokt-network/pocket/shared/crypto"
+
+	"github.com/ProtonMail/go-ecvrf/ecvrf"
+)
+
+const (
+	VRFOutputSize = sha512.Size // See github.com/ProtonMail/go-ecvrf for details
+)
+
+type SecretKey ecvrf.PrivateKey
+type VerificationKey ecvrf.PublicKey
+
+// These are slices rather than arrays in order to more easily comply with the underlying `go-ecvrf/ecvrf` library.
+type VRFProof []byte  // A proof to verify that VRFOutput belongs to a certain publicKey.
+type VRFOutput []byte // Uniformally distributed output that can be normalized to be used in a binomial distribution.
+
+func CreateVRFRandReader(lastBlockHash string, privKey crypto.PrivateKey) (io.Reader, error) {
+	if privKey == nil {
+		return nil, fmt.Errorf("private key cannot be nil")
+	}
+
+	if len(lastBlockHash) < crypto.SeedSize/2 {
+		return nil, fmt.Errorf("the last block hash must be at least %d bytes in length", crypto.SeedSize/2)
+	}
+
+	privKeySeed := privKey.Seed()[:crypto.SeedSize/2]
+	blockHashSeed := lastBlockHash[:crypto.SeedSize/2]
+
+	seed := make([]byte, crypto.SeedSize)
+	copy(seed, privKeySeed)
+	copy(seed, blockHashSeed)
+
+	return bytes.NewReader(seed), nil
+}
+
+// TODO(in this commit): Should we return pointers or values here?
```

Comment:
pointers seem appropriate to me, please remove TODOs if this is the quorum

Reply:
Done.